ra/registration-authority.go

```diff
@@ -121,23 +121,38 @@ func validateEmail(ctx context.Context, address string, resolver bdns.DNSResolve
 	domain := strings.ToLower(splitEmail[len(splitEmail)-1])
 	var resultMX []string
 	var resultA []net.IP
-	resultMX, err = resolver.LookupMX(ctx, domain)
-	if err == nil && len(resultMX) == 0 {
-		resultA, err = resolver.LookupHost(ctx, domain)
-		if err == nil && len(resultA) == 0 {
-			return &probs.ProblemDetails{
-				Type:   probs.InvalidEmailProblem,
-				Detail: emptyDNSResponseDetail,
-			}
-		}
+	var errMX, errA error
```

Comment:
Instead of having a bunch of predefined vars and a `WaitGroup` why not just use channels? You could do something along the lines of

```
errors := make(chan error, 2)
records := int64(0)

for _, f := range []func(context.Context, string) ([]interface{}, error){resolver.LookupMX, resolver.LookupA} {
	go func() {
		r, err := f(ctx, domain)
		if err != nil {
			errors <- err
			return
		}
		sync.AddInt64(&records, len(r))
	}()
}

if len(errors) > 0 || records == 0 {
	// do errory things
}
```